src/Symfony/Bundle/SecurityBundle/Security/ContextAwareFirewallMapInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security;
+
+use Symfony\Component\Security\Http\FirewallMapInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * This interface must be implemented by firewall maps that are able to return a
+ * context.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+interface ContextAwareFirewallMapInterface extends FirewallMapInterface
```

Comment:
The problem is that the `FirewallMapInterface` comes from the Security component, and is implemented in the [`FirewallMap` class of the component](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Http/FirewallMap.php) in which having the `ContextAwareFirewallMapInterface::getContext()` would not really make sense. WDYT?